seqr/views/utils/terra_api_utils.py

```diff
@@ -192,10 +214,43 @@ def user_get_workspace_acl(user, workspace_namespace, workspace_name):
               "canCompute": true
             }
           }
+
     """
     path = "api/workspaces/{0}/{1}/acl".format(workspace_namespace, workspace_name)
     try:
         return _user_anvil_call('get', path, user).get('acl', {})
+    # Exceptions are handled to return an empty result for the users who have no permission to access the acl
     except (TerraNotFoundException, PermissionDenied) as et:
         logger.warning(str(et))
         return {}
+
+
+def add_service_account(user, workspace_namespace, workspace_name):
+    """
+    Add the seqr service account to the workspace on AnVIL.
+
+    The user must have the "can share" privilege for the workspace.
+
+    :param user: the seqr user object
+    :param workspace_namespace: namespace or billing project name of the workspace
+    :param workspace_name: name of the workspace on AnVIL. The name will also be used as the name of project in seqr
+    :return: Success: True, Fail: False
+    """
+    old_acl = user_get_workspace_acl(user, workspace_namespace, workspace_name)
+    service_account = old_acl.get(SERVICE_ACCOUNT_FOR_ANVIL)
+    if service_account and not service_account['pending']:
+        return True
+    acl = [
+             {
+               "email": SERVICE_ACCOUNT_FOR_ANVIL,
+               "accessLevel": "READER",
+               "canShare": False,
+               "canCompute": False
+             }
+          ]
+    path = "api/workspaces/{0}/{1}/acl".format(workspace_namespace, workspace_name)
+    r = _user_anvil_call('patch', path, user, data=json.dumps(acl))
+    if not (r['usersUpdated'] and r['usersUpdated'][0]['email'] == SERVICE_ACCOUNT_FOR_ANVIL):
+        message = 'Failed to grant seqr service account access to the workspace {}/{}'.format(workspace_namespace, workspace_name)
+        logger.warning(message)
```

Comment:
As mentioned above, you do not need to explicitly log warnings on uncaught exceptions, we have middleware that will handle the logging for us